src/code42cli/click_ext/types.py

```diff
@@ -0,0 +1,39 @@
+import json
+import re
+
+import click
+
+
+class JSONFileOrString(click.File):
+    """Declares a parameter to be either a JSON string or a path to a file containing parsable
+    JSON.
+
+    If the argument's first non-whitespace character is a "{", assumes it's JSON and tries to parse
+    it, otherwise assumes it's a filename and tries to open file then parse contents as JSON.
+    """
+
+    def __init__(self):
+        super().__init__("r")
+
+    def convert(self, value, param, ctx):
+        if re.match(
+            r"\s*\{", value
+        ):  # assume JSON string since it begins with a bracket
+            try:
```

Comment:
these two try/except blocks can merged into a single function that take a string argument (either `value` or file.read()`). It's okay if the exception message is the same for both of those cases, imo.

Reply:
I separated the file reading from the json parsing which cleaned this logic up.